src/coreclr/vm/jitinterface.cpp

```diff
@@ -9849,6 +9846,125 @@ CorInfoUnmanagedCallConv CEEInfo::getUnmanagedCallConv(CORINFO_METHOD_HANDLE met
     return result;
 }
 
+/*********************************************************************/
+
+    // return the entry point calling convention for any of the following
+    // - a P/Invoke
+    // - a method marked with UnmanagedCallersOnly 
+    // - a function pointer with the CORINFO_CALLCONV_UNMANAGED calling convention.
+CorInfoCallConvExtension CEEInfo::getEntryPointCallConv(CORINFO_METHOD_HANDLE method, CORINFO_SIG_INFO* callSiteSig)
+{
+    CONTRACTL {
+        THROWS;
+        GC_TRIGGERS;
+        MODE_PREEMPTIVE;
+    } CONTRACTL_END;
+
+    CorInfoCallConvExtension callConv = CorInfoCallConvExtension::Managed;
+
+    JIT_TO_EE_TRANSITION();
+
+    _ASSERTE(method != nullptr || callSiteSig != nullptr);
+
+    if (method)
+    {
+        MethodDesc* pMD = GetMethod(method);
+        if(pMD->IsNDirect())
+        {
+            EX_TRY
+            {
+                PInvokeStaticSigInfo sigInfo(pMD, PInvokeStaticSigInfo::NO_THROW_ON_ERROR);
+
+                switch (sigInfo.GetCallConv()) {
+                    case pmCallConvCdecl:
+                        callConv = CorInfoCallConvExtension::C;
+                        break;
+                    case pmCallConvStdcall:
+                        callConv = CorInfoCallConvExtension::Stdcall;
+                        break;
+                    case pmCallConvThiscall:
+                        callConv = CorInfoCallConvExtension::Thiscall;
+                        break;
+                    default:
+                        callConv = CorInfoCallConvExtension::Managed;
+                        break;
+                }
+            }
+            EX_CATCH
+            {
+                callConv = CorInfoCallConvExtension::Managed;
+            }
+            EX_END_CATCH(SwallowAllExceptions)
+        }
+#if !defined(TARGET_X86)
+        else if (pMD->HasUnmanagedCallersOnlyAttribute())
```

Comment:
I think this should be assert. The JIT should only calls this for PInvokes (handled above) and HasUnmanagedCallersOnlyAttribute methods.

Reply:
I'm not sure of a good way to condition the callers (which are in the JIT) to only call this when the method is a P/Invoke or `HasUnmanagedCallersOnlyAttribute()` returns true.

Reply:
For PInvoke calls, there is `CORINFO_FLG_PINVOKE` flag.
For unmanaged calli, we should have just one `UNMANAGED` call conv in `CORINFO_SIG_INFO` and get rid of the other unmanaged calling conventions from there. This new API that you are introducing would be the one and only proper way to get the unmanaged calling convention details.
For reverse PInvoke, there is `CORJIT_FLAG_REVERSE_PINVOKE` flag.